Rename Archive Endpoint to Zip endpoint

SaravShah opened the pull request:
I can squash all of the commits if needed but 
First Commit: Migration and schema change
Second Commit: renames in app/ dir
Third Commit: renames in spec/ dir
Fourth Commit: renames in config/ and seeds

closes #975 


Edit need shared_config reviewed too:
https://github.com/sul-dlss/shared_configs/pull/620
https://github.com/sul-dlss/shared_configs/pull/621

ndushay commented on db/migrate/20180712203748_rename_archive_endpoint_to_zip_endpoint.rb:
```diff
@@ -0,0 +1,18 @@
+
+    add_index :preservation_policies_zip_endpoints, :zip_endpoint_id, name: "index_pres_policies_zip_endpoints_on_zip_endpoint_id"
+    add_index :preservation_policies_zip_endpoints, :preservation_policy_id, name: "index_pres_policies_zip_endpoints_on_pres_policy_id"
+
```
why all this whitespace?  (it doesn't matter and it is not worth adjusting unless you have to adjust other stuff)

SaravShah commented:
- removed line 80 or so in spec/models/endpoint_spec the let statement. but i think the test should stay. 

- results_recorder: changed endpoint to zip_endpoint where needed!

- job queue names -- changed to zip_endpoint_events ....will have a shared config PR up soon

- Readme has to do with the `moab_storage_root` ... so leaving it for now.

Rebased and refactored other comments @ndushay  left

ndushay commented on app/models/preserved_object.rb:
```diff
@@ -22,8 +22,24 @@ class PreservedObject < ApplicationRecord
     # Ticket: 920
   end
+
+  # given a version, create any PreservedCopy records for that version which don't yet exist for zip
```
ZippedMoabVersion instead of PreservedCopy?  Could just mention this in the PreservedCopy rename ticket instead.

ndushay replied:
put a note in #976

jmartin-sul replied:
actually, my guess is this is an errant chunk of code that got put back in (erroneously) upon rebase.  i think this method should disappear from PO entirely, as that functionality now lives here:  https://github.com/sul-dlss/preservation_catalog/blob/1d32d8f88ab4d89d453d1153003322f48ba4350c/app/models/preserved_copy.rb#L73-L88

SaravShah replied:
whoops! the rebase was messy...

SaravShah commented:
@ndushay @jmartin-sul  thanks for finding that catch. re-pushed.
